libp2p/protocols/identify.nim

```diff
@@ -95,8 +105,11 @@ proc decodeMsg*(buf: seq[byte]): Option[IdentifyInfo] =
   let r5 = pb.getField(5, protoVersion)
   let r6 = pb.getField(6, agentVersion)
 
+  let r8 = pb.getSignedPeerRecord(8, signedPeerRecord)
+
   let res = r1.isOk() and r2.isOk() and r3.isOk() and
-            r4.isOk() and r5.isOk() and r6.isOk()
+            r4.isOk() and r5.isOk() and r6.isOk() and
+            r8.isOk()
```

Comment:
Aren't signed records optional? We shouldn't fail if it's absent.

Reply:
`isOk` means "valid", to check if it's here you have to do `r8.get()` (which is done below)

Something missing will still be valid

This is a bit subtle, and why I added `getMandatoryField` a few weeks ago, to make this more readable

Reply:
Perhaps in future we could also add a `hasField`, like some protobuf implementations?

Reply:
I think an ideal UX would be to just have `hasField` & `getMandatoryField` (which we could rename to `getField`)
But maybe that leads to double-parsing which can be slightly sub optimal?

Reply:
And both could potentially fail because of invalid pb

Reply:
Moved to a new issue: https://github.com/status-im/nim-libp2p/issues/653